api/src/main/java/jakarta/json/stream/JsonParser.java

```diff
@@ -475,6 +475,17 @@ default public void skipObject() {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Gets the event in course.
+     *
+     * @return the current event.
+     
+     * @since 2.1
+     */
+    default public Event currentEvent() {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
I'd move this next to other event related methods (likely after the `next()`) and improved javadoc to look more consistent with the rest. Ie return could be updated to `@return the event for the current parsing state`